lib/GithubUtils.js

```diff
@@ -60,4 +65,59 @@ export default class GithubUtils {
             throw new Error('Unable to find StagingDeployCash issue with correct data.');
         }
     }
+
+    /**
+     * Generate a comparison URL between two versions following the semverLevel passed
+     *
+     * @param {String} repoSlug - The slug of the repository: <owner>/<repository_name>
+     * @param {String} tag - The tag to compare first the previous semverLevel
+     * @param {String} semverLevel - The semantic versioning MAJOR, MINOR, PATCH and BUILD
+     * @return {Promise} the url generated
+     */
+    generateVersionComparisonURL(repoSlug, tag, semverLevel) {
+        return new Promise((resolve, reject) => {
+            const getComparisonURL = (previousTag, currentTag) => (
+                `${EXPENSIFY_CASH_URL}/compare/${previousTag}...${currentTag}`
+            );
+
+            const [repoOwner, repoName] = repoSlug.split('/');
+            const tagSemver = semverParse(tag);
+
+            return this.octokit.repos.listTags({
+                owner: repoOwner,
+                repo: repoName,
+            })
+                .then(githubResponse => githubResponse.data.some(({name: repoTag}) => {
+                    if (semverLevel === 'MAJOR'
+                        && semverSatisfies(repoTag, `<${tagSemver.major}.x.x`)
+                    ) {
+                        resolve(getComparisonURL(repoTag, tagSemver));
+                        return true;
+                    }
+
+                    if (semverLevel === 'MINOR'
+                        && semverSatisfies(repoTag, `<${tagSemver.major}.${tagSemver.minor}.x`)
+                    ) {
+                        resolve(getComparisonURL(repoTag, tagSemver));
+                        return true;
+                    }
+
+                    if (semverLevel === 'PATCH'
+                        && semverSatisfies(repoTag, `<${tagSemver}`)
+                    ) {
+                        resolve(getComparisonURL(repoTag, tagSemver));
+                        return true;
+                    }
+
+                    if (semverLevel === 'BUILD'
+                        && semverSatisfies(repoTag, `<=${tagSemver.major}.${tagSemver.minor}.${tagSemver.patch}`)
+                    ) {
+                        resolve(getComparisonURL(repoTag, tagSemver));
+                        return true;
+                    }
+                    return false;
+                }))
+                .catch(githubError => reject(core.setFailed(githubError)));
```

Comment:
I think this code should be unaware of the Github Actions it's meant to run in. So instead of using `core.setFailed` I would just do:

```suggestion
                .catch(githubError => reject(githubError));
```